### DISCUSSION TITLE
Making a public user op

### DISCUSSION THREAD

**Discussion by positlabs** [cables-gl/cables_docs#199] [Answers & Questions] [answered]
I took the webcam v2 patch from the dev environment and made a custom user op from it since I plan on using it a lot. It works perfectly except that other users can't load it in the editor, even though I published the project. Is there some way to make that patch public? 

https://cables.gl/edit/6kk1nZ

I also tried moving that code into a custom op, but there was an issue with that approach (vague errors in the console). After making the custom op, I can't save the project.

**Comment by steam0r** [comment, marked as answer]
user-op definitely is the right approach.

viewing the patch on the patchpage works for all users, logged in or not: https://cables.gl/p/6kk1nZ

the editor only works for for logged in users, even if they are not collaborators in your patch. 

this will get better and easier in the future.

**Comment by positlabs** [comment]
Got it, that makes sense. I was testing in incognito to see how it would behave. In this case, showing a clearer error message that asks the user to log in would be helpful.

**Reply by steam0r** [reply]
agree, as said we will rework this anyhow. thanks for the input!
